src/ModelContextProtocol.Core/Authentication/ClientOAuthProvider.cs

```diff
@@ -721,23 +792,22 @@ private async Task<ProtectedResourceMetadata> ExtractProtectedResourceMetadata(H
 
         foreach (var part in parameters.Split(','))
         {
-            string trimmedPart = part.Trim();
+            var trimmedPart = part.AsSpan().Trim();
             int equalsIndex = trimmedPart.IndexOf('=');
 
             if (equalsIndex <= 0)
             {
                 continue;
             }
 
-            ReadOnlySpan<char> key = trimmedPart.AsSpan().Slice(0, equalsIndex).Trim();
+            var key = trimmedPart[..equalsIndex].Trim();
 
             if (key.Equals(parameterName, StringComparison.OrdinalIgnoreCase))
             {
-                ReadOnlySpan<char> value = trimmedPart.AsSpan(equalsIndex + 1).Trim();
-
-                if (value.Length > 0 && value[0] == '"' && value[value.Length - 1] == '"')
+                var value = trimmedPart[(equalsIndex + 1)..].Trim();
+                if (value.Length > 0 && value[0] == '"' && value[^1] == '"')
```

Comment:
What should happen if value is just a single `"`?

Reply:
As before the PR, `param=",` results in an `ArgumentOutOfRangeException`. Not great. I don't think there's a built-in API that robustly parses WWW-Authenticate parameters though, and parsing the parameters by hand while accounting for things like `param=","` and `param="\""` deserves it's own PR. I filed an issue tracking fixing this at #1088.